selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/RecreateUpdate.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.hotupdate;
+
+import static java.lang.Integer.*;
+import static org.testng.Assert.*;
+
+import com.google.inject.Inject;
+import java.io.IOException;
+import org.eclipse.che.commons.json.JsonHelper;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.TestGroup;
+import org.eclipse.che.selenium.core.executor.OpenShiftCliCommandExecutor;
+import org.eclipse.che.selenium.core.provider.CheTestApiEndpointUrlProvider;
+import org.eclipse.che.selenium.core.requestfactory.CheTestAdminHttpJsonRequestFactory;
+import org.eclipse.che.selenium.core.utils.WaitUtils;
+import org.eclipse.che.selenium.core.utils.process.ProcessAgent;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.CheTerminal;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+@Test(groups = {TestGroup.OSIO, TestGroup.MULTIUSER})
```

Comment:
`TestGroup.OSIO` is for ospenshift.io specific tests, whereas this test is OpenShift specific one.